src/ray/raylet/node_manager.cc

```diff
@@ -1659,7 +1665,8 @@ void NodeManager::HandleReturnWorker(const rpc::ReturnWorkerRequest &request,
       if (worker->IsBlocked()) {
         HandleDirectCallTaskUnblocked(worker);
       }
-      cluster_task_manager_->ReturnWorkerResources(worker);
+      cluster_task_manager_->ReleaseWorkerResources(worker);
+      cluster_task_manager_->ClearWorkerBacklog(worker_id);
```

Comment:
I don't think this is the right place to clear the worker backlog. This looks like it gets called when a lease is returned to the raylet, but I think we actually want to clear the backlog if the lease *requestor* is killed, etc.

Reply:
Oh yea, good catch. It's not returning itself but leased workers.